qutebrowser/mainwindow/statusbar/command.py

```diff
@@ -54,14 +54,14 @@ class Command(misc.MinimalLineEditMixin, misc.CommandLineEdit):
     show_cmd = pyqtSignal()
     hide_cmd = pyqtSignal()
 
-    def __init__(self, win_id, parent=None):
-        misc.CommandLineEdit.__init__(self, parent)
+    def __init__(self, *, win_id, private, parent=None):
+        misc.CommandLineEdit.__init__(self, parent=parent)
         misc.MinimalLineEditMixin.__init__(self)
         self._win_id = win_id
-        command_history = objreg.get('command-history')
-        self.history.handle_private_mode = True
-        self.history.history = command_history.data
-        self.history.changed.connect(command_history.changed)
+        if not private:
+            command_history = objreg.get('command-history')
+            self.history.history = command_history.data
+            self.history.changed.connect(command_history.changed)
```

Comment:
I've thought about both before, but I think it doesn't warrant the complexity (and possibility for mistakes) - we used to have no command history at all with private browsing turned on, and nobody complained (though it confused some people).